src/storage/engine/mod.rs

```diff
@@ -156,21 +164,43 @@ pub struct Statistics {
     pub seek_for_prev: usize,
 }
 
-impl Statistics {
+impl CFStatistics {
     #[inline]
     pub fn total_op_count(&self) -> usize {
         self.get + self.next + self.prev + self.seek + self.seek_for_prev
     }
 
-    // Calculate how the operation performs.
-    #[inline]
-    pub fn inefficiency(&self) -> f64 {
-        let total_op_cnt = self.total_op_count();
-        if total_op_cnt == 0 {
-            0f64
-        } else {
-            (total_op_cnt as f64 - self.processed as f64) / total_op_cnt as f64
-        }
+    pub fn details(&self) -> Vec<(&str, usize)> {
```

Comment:
If we call details too frequently, this may be not efficiency. I suggest holding the metrics in the local statistics and then flush to the global metrics sometimes, not everytime.